src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/Diagnostics/RazorSourceGeneratorResources.resx

```diff
@@ -177,4 +177,10 @@
   <data name="MetadataReferenceNotProvidedMessage" xml:space="preserve">
     <value>Expected a valid MetadataReference, but found none.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="DuplicateRazorFileIncludedTitle" xml:space="preserve">
+    <value>Razor file was included twice</value>
+  </data>
+  <data name="DuplicateRazorFileIncludedMessage" xml:space="preserve">
+    <value>File '{0}' was included in the build twice. This will cause generator build issues.</value>
```

Comment:
Nit: Should this be more specific and say "was included as an additional file" rather than just "in the build"? Including it in the build twice isn't actually a problem if the items have different item types, eg EmbeddedResource and AdditionalFiles, right?

Also "more than once" rather than "twice" because it could actually be three times 😛